test/compile/data/transform.test.ts

```diff
@@ -0,0 +1,55 @@
+import {assert} from 'chai';
+import {AggregateNode} from '../../../src/compile/data/aggregate';
+import {BinNode} from '../../../src/compile/data/bin';
+import {TimeUnitNode} from '../../../src/compile/data/timeunit';
+import {CalculateNode, FilterNode, parseTransformArray} from '../../../src/compile/data/transforms';
+import {TimeUnit} from '../../../src/timeunit';
+import {parseUnitModel} from '../../util';
+describe('compile/data/transform', () => {
+  describe('parseTransformArray()', () => {
+    it('should return a CalculateNode and a FilterNode', () => {
+      const model = parseUnitModel({
+        data: {values: []},
+        mark: 'point',
+        transform: [{calculate: 'calculate', as: 'as'}, {filter: 'filter'}],
+        encoding: {
+          x: {field: 'a', type: 'temporal', timeUnit: 'month'}
+        }
+      });
+
+      const result = parseTransformArray(model);
+      assert.isTrue(result.first instanceof CalculateNode);
+      assert.isTrue(result.last instanceof FilterNode);
+    });
+
+    it('should return a BinNode node and a TimeUnitNode', () => {
+      const model = parseUnitModel({
+        data: {values: []},
+        mark: 'point',
+        transform: [{bin: true, field: 'field', as: 'as'}, {timeUnit: 'month', field: 'field', as: 'as'}],
```

Comment:
This is a bad example -- why name two different things with the same name (`'as'`) ?